### PR TITLE
Add a script that generates changelog for patch release

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

- Add `./scripts/changelog-for-patch.mjs`
- Extract common functions to `./scripts/utils/changelog`
- Use `changelog-for-patch.mjs` from release script

## Checklist

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: scripts/changelog-for-patch.mjs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    printEntries(entries).join("\n\n"),
+    previousVersion,
+    newVersion,
+    /** isPatch */ true
+  )
+);
+
+function parseArgv() {
+  const argv = minimist(process.argv.slice(2));
+  const previousVersion = argv["prev-version"];
```
Comment (fisker) [inline scripts/changelog-for-patch.mjs:32]: Shouldn't previous version always `require("prettier").version`?

Comment (sosukesuzuki) [reply scripts/changelog-for-patch.mjs:32]: I think that's correct in most cases. However, the release script actually get the previous version in a different way(ref: https://github.com/prettier/prettier/blob/acf9242dd142a042d889ccdac649c367cc76b910/scripts/release/release.js#L20-L24 ). So I think it is safer to be able to specify it as an argument.

Comment (sosukesuzuki) [reply scripts/changelog-for-patch.mjs:32]: @fisker What do you think? I want to merge this for next patch version.

**Review by fisker — APPROVED** [review]
LGTM, should we call prettier to format file?  Maybe it will fail on lint ?

**Review by Raud-pi — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
> should we call prettier to format file? Maybe it will fail on lint ?

We call prettier at with `--write` https://github.com/prettier/prettier/blob/15b112b9752d107db5477b3b629cd8919cde4cb7/scripts/release/steps/update-changelog.js#L80-L83

**Pull request merged**